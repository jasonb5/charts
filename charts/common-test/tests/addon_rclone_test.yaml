suite: test rclone addon
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should pass
    set:
      workload: deployment

      persistence:
        config:
          enabled: true
          type: emptydir

      addons:
        rclone:
          enabled: true

          env:
            CRON: '0 */15 * * *'
            SOURCE: /config
            DESTINATION: /backup
            FLAGS: --checksum

          persistence:
            config:
              enabled: true
              type: emptydir

            backup:
              enabled: true
              type: nfs
              server: 192.168.1.1
              path: /mnt/user/backup
              subPath: app
    asserts:
      - matchSnapshot:
          path: spec.template.spec.containers
        documentIndex: 0
      - hasDocuments:
          count: 2
