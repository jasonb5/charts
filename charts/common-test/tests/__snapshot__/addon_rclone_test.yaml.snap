should pass:
  1: |
    - image: busybox:0.1.0
      livenessProbe:
        tcpSocket:
          port: default
      name: common-test
      ports:
        - containerPort: null
          name: default
          protocol: TCP
      readinessProbe:
        tcpSocket:
          port: default
      volumeMounts:
        - mountPath: /config
          name: config
    - env:
        - name: CRON
          value: 0 */15 * * *
        - name: DESTINATION
          value: /backup
        - name: FLAGS
          value: --checksum
        - name: SOURCE
          value: /config
      image: ghcr.io/jasonb5/rclone:0.1.0
      name: rclone
      volumeMounts:
        - mountPath: /backup
          name: backup
          subPath: app
        - mountPath: /config
          name: config
